src/libraries/System.Private.Xml/src/System/Xml/Xsl/QIL/SubstitutionList.cs

```diff
@@ -46,12 +47,12 @@ public void RemoveLastSubstitutionPair()
         /// </summary>
         /// <param name="n">the node to replace</param>
         /// <returns>null if no replacement is found</returns>
-        public QilNode FindReplacement(QilNode n)
+        public QilNode? FindReplacement(QilNode n)
         {
             Debug.Assert(_s.Count % 2 == 0);
             for (int i = _s.Count - 2; i >= 0; i -= 2)
                 if (_s[i] == n)
-                    return (QilNode)_s[i + 1];
+                    return (QilNode)_s[i + 1]!;
```

Comment:
this will return not null, the only reason it needs `!` is because it uses ArrayList. Returning nullable being ok is incidental here. I assume this will eventually become typed array/list.